src/python/pants/util/meta.py

```diff
@@ -23,7 +24,7 @@ def __call__(cls, *args: Any, **kwargs: Any) -> Any:
     return cls.instance
 
 
-T = TypeVar("T")
+_T = TypeVar("_T")
```

Comment:
Good change! The main reason I started making type vars private is to avoid autocomplete suggesting it in other files. Other files shouldn't really use this type var.